src/bramble/client/ProjectStats.js

```diff
@@ -0,0 +1,258 @@
+    _fs.unlink = function(pathname, callback){
+        // call original Filer.unlink
+        _innerUnlink.call(_fs, pathname, function(err){
+            // only update cache once original was successful
+            if(!err){
+                // stop keeping track of deleted file
+                delete _cache[pathname];
+            }
+            
+            // call callback regardless
+            callback(err);
+        });
+    };
+
+    // original writeFile
+    var _innerWriteFile = _fs.writeFile;
+    // overwrite original writeFile and add bookkeeping of project state and call original writeFile.
+    _fs.writeFile = function(filename, data, options, callback){
+        function update(){
+            //only check once _root is established
+            if(_root){
+                // owerwrite or add a new size for a given filename
+                _cache[filename] = data.length;
+            }
+        };
+
+        if(arguments.length === 3){
+            // callback is now assigned to optins
+            _innerWriteFile.call(_fs, filename, data, function(err){
+                // only update cache once original was successful
+                if(!err){
+                    update();
+                }
+                
+                // call callback regardless
+                options(err);
+            });
+        } else {
+            _innerWriteFile.call(_fs, filename, data, options, function(err){
+                // only update cache once original was successful
+                if(!err){
+                    update();
+                }
+                
+                // call callback regardless
+                callback(err);
+            });
+        }
+    };
+
+    // original rename
+    var _innerRename = _fs.rename;
+    // overwrite original rename and add bookkeeping of project state and call original rename.
+    // this is essential because we don't want to lose track of file's size for renamed files.
+    _fs.rename = function(oldPath, newPath, callback){
+        _innerRename.call(_fs, oldPath, newPath, function(err){
+            // only update cache once original was successful
+            if(!err){
+                _cache[newPath] = _cache[oldPath];
+                delete _cache[oldPath];
+            }
+            
+            // call callback regardless
+            callback(err);
+        });
+    };
+
+    // returns true/false depending on whether root/index.html exists
+    ProjectStats.hasIndexFile = function() {
+        // Loop through all paths, return true if one of them is root/index.html
+        return !!(Object.keys(_cache)).find(function(path) {
+            return path === _Path.join(_root, "index.html");
+        });
+    };
+
+    // returns an array of filenames for specified extension
+    // adds period if not provided
+    ProjectStats.usesExtension = function(fileExtensions, callback) {
```

Comment:
Let's drop this for now, I don't know exactly how we'll use it, so I don't think we should include yet.